File: 0053-maximum-subarray/0053-maximum-subarray.py
class Solution(object):
    def maxSubArray(self, nums):
        current_sum = nums[0]
        max_sum = nums[0]
        
        for i in range(1, len(nums)):
            # Either extend the previous subarray or start new
            current_sum = max(nums[i], current_sum + nums[i])
            
            # Update global maximum
            max_sum = max(max_sum, current_sum)
        
        return max_sum

File: 0053-maximum-subarray/0053-maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int current_sum = nums[0];
        int max_sum = nums[0];
        
        for (int i = 1; i < nums.size(); i++) {
            // Either start new subarray or continue previous one
            current_sum = max(nums[i], current_sum + nums[i]);
            
            // Update global maximum
            max_sum = max(max_sum, current_sum);
        }
        
        return max_sum;
    }
};